README.md

```diff
@@ -16,3 +16,53 @@ Also give a theoretical argument for why X could or could not be correct, based
 on the complexity of the general sorting problem we covered in class.
 
 Add your answers to this markdown file.
+
+### Verifying X
+
+There are two key aspects that I would want to verify. The first is that it
+is in fact a sorting algorithm for arbitrary elements. The other is that it 
+is in fact in $O(n)$ time, or at least "faster than any known general sorting 
+algorithm."
+
+To verify the first claim, I would write some property-based tests. This code would
+generate a random list of integers and ensure that for any given list, (and all
+of its permutations) the algorithm outputs a sorted list. The key part of this is that
+all permutations must be returned sorted because the main concern is that a sorting algorithm
+with linear time complexity wouldn't be able to possibly generate all orderings of a list. I 
+would also want to verify that this works with other datatypes as well. We don't know how 
+elements are being compared, but we know that two elements are being compared at a time. It's
+possible that the comparison method wouldn't work as well with datatypes like strings or
+a type of object that requires something like a lexicographic ordering that requires attributes
+of an object to be compared based on a hierarchy. 
+
+For the second claim, it would be difficult to disprove if the program scaled linearly or not without
+being able to see the source code. We could verify whether or not it is faster than other algorithms 
+though. I would approach this issue by testing the run time of the program with varied input size. 
+I would do this in a similar way to the comparisons in the sorting comparison lecture, comparing to 
+quicksort. My main goal see how the program behaves when the input gets large. If it consistently 
+ran significantly faster than quicksort, there would be evidence to suggest that the time complexity 
+is $O(n)$, or at least faster than any algorithm we know of. This test is not perfect, unfortunately.
+We don't know the memory complexity or additional constants of the time complexity. In other words, 
+we can't completely disprove the claim without knowing the implementation, but we would be able 
+to verify if it is "faster than any known general sorting algorithm," in practice. 
```

Comment:
No, we can verify whether the algorithm scales linearly in practice. It's not necessary to compare it to any other algorithm for that.